src/modules/DNS.rb

```diff
@@ -246,24 +246,16 @@ def default_dhcp_hostname
     end
 
     def ReadHostname
-      fqhostname = ""
       # In installation (standard, or AutoYaST one), prefer /etc/install.inf
       # (because HOSTNAME comes with netcfg.rpm already, #144687)
       if (Mode.installation || Mode.autoinst) && FileUtils.Exists("/etc/install.inf")
         fqhostname = read_hostname_from_install_inf
       end
 
-      fqhostname = read_hostname_from_etc if fqhostname.empty?
+      # reads setup from /etc/HOSTNAME, returns a default if nothing found
+      fqhostname = Hostname.CurrentFQ if fqhostname.nil? || fqhostname.empty?
```

Comment:
The docs for `Hostname.CurrentFQ` say 
> Retrieve currently set fully qualified hostname
> (uses hostname --fqdn)

It does that, but in fact it falls back to /etc/HOSTNAME and to linux.$tld. Notably it does not use /etc/hostname like the deleted method. So there may be a different bug hidden in there.
https://github.com/yast/yast-yast2/blob/02ec574edeaba563f0e52a9b3d5cf58e0fe13d59/library/types/src/modules/Hostname.rb#L154